pkg/reconciler/gc/v2/gc.go

```diff
@@ -61,6 +61,10 @@ func Collect(
 	})
 
 	for _, rev := range revs[gcSkipOffset:] {
+		preserve := rev.ObjectMeta.Annotations[serving.RevisionPreservedAnnotationKey]
+		if preserve == "true" {
+			continue
+		}
```

Comment:
Here an in interesting question is: should we first filter than apply `gcSkipOffset` limit, since this counts as a revision for minRevision to retain as well?
cc @dprotaso @mattmoor

Reply:
I think we want it to not count towards the min/max (and treat it as if it were an actively referenced revision even if it is not). The settings for automatic-gc say to the user "how many extra should we keep automatically keep around" for potential rollback, I don't think we'd want it to cut into that if the user is choosing to manage these manually - they could set min to 0 if, for example, they intend to manage everything manually.

I intend for the code layout to change a little bit as we integrate the new v2 settings which might give it a more clear place to live and gets rid of the offset: https://github.com/knative/serving/pull/8621

Reply:
We need to make some more meaningful settings like whaught said to meet this question.